src/Producer/SyncProcess.php

```diff
@@ -134,12 +134,15 @@ protected function convertMessage($data)
 
             $topicMeta = $topicInfos[$value['topic']];
             $partNums  = array_keys($topicMeta);
-            shuffle($partNums);
-            $partId = 0;
-            if (! isset($value['partId']) || ! isset($topicMeta[$value['partId']])) {
-                $partId = $partNums[0];
+            if (isset($value['key']) && trim($value['key'])) {
```

Comment:
Extracting this logic to a method and using early returns will make things way easier to understand, could you please do that?